pkg/services/pluginsintegration/config/config.go

```diff
@@ -35,6 +35,9 @@ func ProvideConfig(settingProvider setting.Provider, grafanaCfg *setting.Cfg, fe
 		allowedAuth,
 		aws.KeyValue("assume_role_enabled").MustBool(grafanaCfg.AWSAssumeRoleEnabled),
 		aws.KeyValue("external_id").Value(),
+		aws.KeyValue("session_duration").Value(),
+		grafanaCfg.AWSListMetricsPageLimit,
```

Comment:
`settingsProvider` can be backed by a database instead of a file - which I believe AWS originally wanted so that config could be reloaded on the fly. When not used, the settingsProvider just falls back to the file anyway. That was the original motivation there!

Reply:
Ah, I see! Should I update the new settings to be gotten off `settingsProvider`? It's a little weird because `allowedAuth` is actually being gotten from `grafanaCfg`

Reply:
Yeah TBH I'm not sure we've done a great job here of making it clear what should be happening (and I don't think many know the actual details of `settingsProvider`). I just remember needing to do this for plugin settings originally - but I think any config that people would like to potentially be updated via API/backend by the DB/can change at runtime, should be fetched from the `settingsProvider`!

Reply:
I made it so that the aws settings (including `allowedAuth`) are gotten from the settingsProvider and added a comment so hopefully we remember